src/psyclone/tests/domain/lfric/lfric_extract_driver_creator_test.py

```diff
@@ -0,0 +1,510 @@
+                                "extract", "_post", ("region", "name"))
+    assert (f"Cannot create driver for 'region-name' because:\nLFRic builtin "
+            f"'{name}' is not supported" in out)
+
+
+# ----------------------------------------------------------------------------
+def test_lfric_driver_array_of_fields():
+    '''Check that array accesses are created correctly, which is required
+    for builtins. E.g. the following code needs to be created:
+        do df = loop0_start, loop0_stop, 1
+            f2(df) = a + f1(df)
+        enddo
+    As opposed to a normal kernel (where the whole field is passed in),
+    here we need explicit array accesses.
+    '''
+
+    _, invoke = get_invoke("15.1.8_a_plus_X_builtin_array_of_fields.f90",
+                           API, dist_mem=False, idx=0)
+    dep = DependencyTools()
+    input_list, output_list = dep.get_in_out_parameters(invoke.schedule)
+    driver_creator = LFRicExtractDriverCreator()
+
+    driver = driver_creator.\
+        get_driver_as_string(invoke.schedule, input_list, output_list,
+                             "extract", "_post",
+                             region_name=("region", "name"))
+
+    assert "f2(df) = a + f1(df)" in driver
+
+
+# ----------------------------------------------------------------------------
+@pytest.mark.usefixtures("change_into_tmpdir")
+def test_lfric_driver_extract_some_kernels_only():
+    '''Test that we can extract only some kernels of an invoke, but still
+    get the right loop boundaries (which are dependent on the index
+    of the kernels in the tree). This test can potentially be removed
+    when TODO #1731 is done.'''
+
+    _, invoke = get_invoke("4.5.2_multikernel_invokes.f90", API,
+                           dist_mem=False, idx=0)
+
+    extract = LFRicExtractTrans()
+    extract.apply(invoke.schedule.children[2],
+                  options={"create_driver": True,
+                           "region_name": ("field", "test")})
+    code = str(invoke.gen())
+
+    # We only extract the third loop, which uses the index '2' for
+    # loop boundaries. So none of the previous loop indices should
+    # be in the extract code:
+    assert "PreDeclareVariable(\"loop0_start\", loop0_start)" not in code
+    assert "PreDeclareVariable(\"loop1_start\", loop1_start)" not in code
+    assert "PreDeclareVariable(\"loop2_start\", loop2_start)" in code
+    assert "PreDeclareVariable(\"loop2_stop\", loop2_stop)" in code
+
+    filename = "driver-field-test.f90"
+    with open(filename, "r", encoding='utf-8') as my_file:
+        driver = my_file.read()
+
+    # Make sure the driver does not have any information about other
+    # kernels added, and that is uses index 2 for loop boundaries.
+    assert "loop0_start" not in driver
+    assert "loop1_start" not in driver
+    assert "ReadVariable('loop2_start', loop2_start)" in driver
+    assert "ReadVariable('loop2_stop', loop2_stop)" in driver
+
```

Comment:
It would be great to test compilation of the driver but I guess we're quite a long way from being able to do that? (While I'm here, LFRic itself is now split into a library and an application which will help us one day.)

Reply:
I just tried, it mostly works, except:
1) we need to additionally specify the path to the read-kernel library
2) we also need to compile the kernel called first.

Given that #1991 will remove these requirements by inlining all of this, I suggest we postpone this to #1991.

Reply:
I am happy to report that the drivers in the tests also do compile, I've just added this to #1991.